src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -526,11 +526,11 @@ public function addAllowedValues($option, $allowedValues)
         }
 
         if (!isset($this->allowedValues[$option])) {
-            $this->allowedValues[$option] = $allowedValues;
-        } else {
-            $this->allowedValues[$option] = array_merge($this->allowedValues[$option], $allowedValues);
+            $this->allowedValues[$option] = array();
         }
 
+        $this->allowedValues[$option] = array_merge($this->allowedValues[$option], $allowedValues);
```

Comment:
Also, we can remove the array check above and use `(array) $allowedValues` instead, just like `addAllowedTypes()` method.